### PR TITLE
Display Parents and Children on Arbital Pages

### PR DESCRIPTION
**Pull request by darkruby501**
Probability - LessWrong 2024-12-11 18-04-12

Probability - LessWrong 2024-12-11 18-04-50

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1208962760630870) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/tagging/TagPage.tsx [hunk 1]
```diff
@@ -500,6 +503,48 @@ const styles = defineStyles("TagPage", (theme: ThemeType) => ({
...
+    display: 'inline-flex',
+    alignItems: 'center',
+    '&:not(:last-child)::after': {
+      content: '", "',
+      marginRight: '4px',
+    },
+  },
+  linkedTagMore: {
+    color: theme.palette.link.color,
+    // fontStyle: 'italic',
```
Comment (b0b3rt) [inline packages/lesswrong/components/tagging/TagPage.tsx:546]: cleanup

File: packages/lesswrong/components/tagging/TagPage.tsx [hunk 2]
```diff
@@ -607,53 +652,85 @@ interface ArbitalLinkedPage {
...
+          ))}
+          {!isChildrenExpanded && children?.length > 2 && (
+            <div 
+              className={classes.linkedTagMore} 
+              onClick={(e) => {
+                e.preventDefault();
+                e.stopPropagation();
+                setIsChildrenExpanded(true);
+              }}
+              style={{ cursor: 'pointer' }}
```
Comment (b0b3rt) [inline packages/lesswrong/components/tagging/TagPage.tsx:729]: This style probably shouldn't be inline?

File: packages/lesswrong/components/tagging/TagPage.tsx [hunk 3]
```diff
@@ -668,20 +745,19 @@ const ArbitalRelationshipsSmallScreen = ({arbitalLinkedPages}: {arbitalLinkedPag
     return null;
   }
 
-  const { TagsTooltip, HoverPreviewLink } = Components;
+  const { TagsTooltip } = Components;
   const { requirements, teaches } = arbitalLinkedPages;
   
   return (
+    <Components.ContentStyles contentType="tag">
```
Comment (b0b3rt) [inline packages/lesswrong/components/tagging/TagPage.tsx:752]: destructure this from `Components` above (and indent everything inside of it)

File: packages/lesswrong/components/tagging/TagPage.tsx [hunk 4]
```diff
@@ -668,20 +745,19 @@ const ArbitalRelationshipsSmallScreen = ({arbitalLinkedPages}: {arbitalLinkedPag
...
       {requirements.length > 0 && (
         <div className={classes.relationshipRow}>
           <span className={classes.spaceAfterWord}>{'Requires: '}</span>
           {requirements.map((req: ArbitalLinkedPage, i: number) => (
             <span key={req.slug} className={classes.spaceAfterWord}>
               <TagsTooltip tagSlug={req.slug}>
-                <HoverPreviewLink href={tagGetUrl(req)}>
-                  {req.name}
-                </HoverPreviewLink>
+                  <Link to={tagGetUrl(req)}>{req.name}</Link>
```
Comment (b0b3rt) [inline packages/lesswrong/components/tagging/TagPage.tsx:760]: thx

File: packages/lesswrong/components/tagging/TagPage.tsx [hunk 5]
```diff
@@ -1177,6 +1295,9 @@ const TagPage = () => {
           </ContentStyles>
         </div>
       </AnalyticsContext>
+      <AnalyticsContext pageSectionContext="parentsAndChildrenSmallScreenNavigationButtons">
+        <ParentsAndChildrenSmallScreen arbitalLinkedPages={selectedLens?.arbitalLinkedPages ?? undefined} tagOrLensName={selectedLens?.title ??tag.name} />
```
Comment (b0b3rt) [inline packages/lesswrong/components/tagging/TagPage.tsx:1299]: missing space after second coalesce operator

File: packages/lesswrong/lib/collections/helpers/arbitalLinkedPagesField.ts [hunk 6]
```diff
@@ -227,14 +228,35 @@ export function arbitalLinkedPagesField(options: ArbitalLinkedPagesFieldOptions)
...
+      const parentDocumentIds = parentRels.map((rel) => rel.parentDocumentId);
+
+      // Step Y: Fetch child relationships
+      const childRels = await ArbitalTagContentRels.find(
+        {
+          parentDocumentId: docId,
+          type: 'parent-is-parent-of-child',
+        },
+        { projection: { childDocumentId: 1 } }
+      ).fetch();
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/helpers/arbitalLinkedPagesField.ts:256]: x & y?

Also surely these could be parallelized.

Comment (b0b3rt) [reply packages/lesswrong/lib/collections/helpers/arbitalLinkedPagesField.ts:256]: actually why aren't requirementsRels, teachesRels, and parentRels just a single query with `type: { $in: ['parent-is-requirement-of-child', 'parent-taught-by-child', 'parent-is-parent-of-child'] }`?  (In parallel with fetching the childRels.)

Comment (darkruby501) [reply packages/lesswrong/lib/collections/helpers/arbitalLinkedPagesField.ts]: Addressed

**Review by b0b3rt — APPROVED** [review]

**Pull request merged**